### ISSUE TITLE
Bump to rails 6.1

### ISSUE THREAD

**Issue by colinxfleming** [DCAFEngineering/dcaf_case_management#2110]
Thanks for creating an issue! Please fill out this form so we can be sure to have all the information we need, and to minimize back and forth.

* **What are we trying to do?**

It's upgrade season. HOOOOONK

* **What feature or behavior is this required for?**

new hotness; safe rails

* **How could we solve this issue? (Not knowing is okay!)**

* bump gem
* run the rails app upgrade command and incorporate diffs appropriately
* review new feature list and see if there's anything we should be taking advantage of (probably not, but doesn't hurt to check)
* get everything back to green

* **Anything else?**

low priority for right now

**Comment by DarthHater** [comment]
This is almost ready to work on, it is held up by `mongoid`, whomsteve merged rails 6.1 support but it has yet to be released, more info here: https://github.com/mongodb/mongoid/pull/4938 (would pause until this is released)

**Comment by colinxfleming** [comment]
oops, duped this in #2251 - closing in favor of that
